services/src/main/java/org/keycloak/protocol/oidc/mappers/OIDCAttributeMapperHelper.java

```diff
@@ -307,16 +308,32 @@ private static <T> void mapClaim(T token, ProtocolMapperModel mappingModel, Obje
         }
 
         // map value to the other claims map
-        mapClaim(split, attributeValue, jsonObject);
+        mapClaim(split, attributeValue, jsonObject, isMultivalued(mappingModel));
     }
 
-    private static void mapClaim(List<String> split, Object attributeValue, Map<String, Object> jsonObject) {
+    private static void mapClaim(List<String> split, Object attributeValue, Map<String, Object> jsonObject, boolean isMultivalued) {
         final int length = split.size();
         int i = 0;
         for (String component : split) {
             i++;
-            if (i == length) {
+            if (i == length && length == 1 && !isMultivalued) {
```

Comment:
I don't understand the `length == 1` here, is it needed? The attribute should be appended if multi-valued no matter if the claim is `claim` or `complex.claim` (length of the split), am I right?

Reply:
You are right. I have been removed  it.